db/queries/seller.sql

```diff
@@ -0,0 +1,404 @@
+-- name: SellerGetInfo :one
+
+SELECT
+    "seller_name",
+    "image_id",
+    "description",
+    "enabled"
+FROM "shop"
+WHERE "seller_name" = $1;
+
+-- name: SellerUpdateInfo :one
+
+UPDATE "shop"
+SET
+    "image_id" = COALESCE($2, "image_id"),
+    "name" = COALESCE($3, "name"),
+    "description" = COALESCE($4, "description"),
+    "enabled" = COALESCE($5, "enabled")
+WHERE "seller_name" = $1
+RETURNING
+    "seller_name",
+    "image_id",
+    "name",
+    "enabled";
+
+-- name: SellerSearchTag :many
+
+SELECT t."id", t."name"
+FROM "tag" t
+    LEFT JOIN "shop" s ON "shop_id" = s.id
+WHERE
+    s."seller_name" = $1
+    AND t."name" ~* $2
+ORDER BY LENGTH(t."name") ASC
+LIMIT $3;
+
+-- name: HaveTagName :one
+
+SELECT
+    CASE
+        WHEN EXISTS (
+            SELECT *
+            FROM "tag" t
+                LEFT JOIN "shop" s ON "shop_id" = s.id
+            WHERE
+                s."seller_name" = $1
+                AND t."name" = $2
+        ) THEN true
+        ELSE false
+    END;
+
+-- name: SellerInsertTag :one
+
+INSERT INTO
+    "tag" ("shop_id", "name")
+VALUES ( (
+            SELECT s."id"
+            FROM "shop" s
+            WHERE
+                s."seller_name" = $1
+                AND s."enabled" = true
+        ),
+        $2
+    ) ON CONFLICT ("shop_id", "name")
+DO NOTHING
+RETURNING "id", "name";
+
```

Comment:
Already check conflict at [api](https://github.com/x06lan/twp/blob/61ebc6e2d0391a6e23c44bd096f3cb48272267ee/pkg/router/seller.go#L138-L147), is it neccessary to check again?